specs/Standards/Wallets/InjectedWallets.md

```diff
@@ -0,0 +1,369 @@
+# Injected Wallets
+
+## Summary
+
+Standard interface for injected wallets.
+
+## Motivation
+
+dApps are finding it increasingly difficult to support the ever expanding choice of wallets due to their wildly different implementations. While projects such as [Wallet Selector](https://github.com/near/wallet-selector) attempt to mask this problem, it's clear the ecosystem requires a standard that will not only benefit dApps but make it easier for established wallets to support NEAR.
+
+## Rationale and alternatives
+
+At its most basic, a wallet contains key pairs required to sign messages. This standard aims to define an API (based on our learning from [Wallet Selector](https://github.com/near/wallet-selector)) that achieves this requirement through a number of methods exposed on the `window` object.
```

Comment:
@lewis-sqa [Ethereum's JSON-RPC Methods](https://docs.metamask.io/guide/rpc-api.html#ethereum-json-rpc-methods) are generalizable to web wallets as well e.g. [portis wallet](https://docs.portis.io/#/tips-and-tricks?id=custom-eth_sendtransaction-parameters), do we have to make this NEP extension wallet specific by specifying that this API should be exposed on the `window` object and where?

Reply:
This NEP outlines APIs that don't have direct parallels with a web wallet; event emitters and methods that return promises that resolve to the results of direct calls to an injected wallet can (for example) throw errors directly to the dApp from the wallet, whereas a web wallet running in another tab cannot.  I'm all for making this a more generally applicable API if that makes sense, but I don't think we need to block this landing to accomplish that in the future if desirable.